src/redux/rockets/rockets.js

```diff
@@ -20,10 +21,24 @@ export const fetchRocketsAction = () => (dispatch) => {
   );
 };
 
+export const bookRocket = (currentState, id) => (dispatch) => {
+  const newState = currentState.map((rocket) => {
+    if (rocket.id !== id) return rocket;
+    const reservedOrNot = !rocket.reserved;
+    return { ...rocket, reserved: reservedOrNot };
+  });
+  dispatch({
+    type: RESERVE_ROCKET,
+    newState,
+  });
+};
+
```

Comment:
It generally is the reducer, and not the action creator the one that uses the old state to create the next state. Sometimes we added functionality when using asynchronous operations, but this is not the case here.
I didn't find anything against this way of doing things so if you like it this way, go ahead. I just wanted to point this out.
(The usual way would be something like
```javascript
export const bookRocket = (id) => ({
  type: RESERVE_ROCKET,
  id,
});
```
and let the reducer do take care of the logic)